### PR TITLE
Moving mongo url into a secret

### PR DESCRIPTION
**Pull request by liranmauda**
### Changes:
- Removed the value from the statefulset-core
- Added to the sercretDb a url key
- Getting the mongo db url from a secret

### Not In Scope:
- We are not exposing the db secret in the crd yet, That we will do in another PR 

Signed-off-by: liranmauda <liran.mauda@gmail.com>

### CODE REVIEW

**Comment by liranmauda** [comment]
@dannyzaken 
We can divide the secret DB into `Postgres secret DB` and `mongo secret DB`.
Currently, when we have only secret DB not divided, we populate it with data for Postgres and for mongo. 
What do you think?

**Comment by dannyzaken** [comment]
@liranmauda maybe we should have separate secrets since during upgrade and migration we should deploy both DBs

**Comment by jackyalbo** [comment]
I don't really like this differences between mongo and postgres. If we decide that url is inside a secret for some reason than it suppose to be for both. I also don't think it suppose to be the same secre: a secret that we get from the user and a secret we are creating for saving our own created creds. Also not sure we are suppose to create the secret for him. he should create a secret and send it to us in some way.

**Comment by dannyzaken** [comment]
@jackyalbo I agree that the secret should contain all connection details, but there will always be differences. 
this PR is only handling the internal DB secret that is passed to noobaa-core. the external secret that should be created by the user will be an optional secret to connect to an external DB

**Comment by jackyalbo** [comment]
@dannyzaken I talked with @liranmauda and know I understand better that we want an internal secret without connection to the external one, but I do think we need to put all the connection details in this secret, maybe also the db name - to make it as close to mongo's url as possible. what do you think?

**Comment by guymguym** [comment]
@dannyzaken maybe we can check how the mongodb & postgres operators set their credentials and other access information into kubernetes secrets? Maybe aligning to their way is preferred?

**Comment by dannyzaken** [comment]
@guymguym I think this is more relevant to the external secret passed by the user, isn't it?
in any case, for mongo at least, I think this is the relevant page: https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/users.md - it looks like they only store the password in a secret and the user name is passed in the MongoDB CR. they also have a different way to pass TLS certificates.
as for Postgres - there isn't any formal operator and I guess that each operator has its own thing. didn't check it really

**Comment by liranmauda** [comment]
For Mongo operator, We might need also `publicApiKey` and `user` 
* [mongodb.com/kubernetes-operator/](https://docs.mongodb.com/kubernetes-operator/master/tutorial/create-operator-credentials/index.html)

But I think for starters we can use only a URL as it can contain credentials in the form of `mongodb://user:password@mongoserver.com/...`

* https://github.com/noobaa/noobaa-operator/issues/304#issuecomment-683690960

File: pkg/system/reconciler.go [hunk 1]
```diff
@@ -189,7 +192,8 @@ func NewReconciler(
 	r.ServiceS3.Name = "s3"
 	r.ServiceDb.Name = r.Request.Name + "-db"
 	r.SecretServer.Name = r.Request.Name + "-server"
-	r.SecretDB.Name = r.Request.Name + "-db"
+	r.SecretMongoDB.Name = r.Request.Name + "-db"
+	r.SecretPostgresDB.Name = r.Request.Name + "-db"
```
Comment (liranmauda) [inline pkg/system/reconciler.go:196]: Need to change the names to differentiate between the mongo secret to the postgres

**Comment by liranmauda** [comment]
Moving to draft.
After talking with @lallinger-arbeit and @QuintinBecker we might want to drop this effort

**Comment by lallinger-arbeit** [comment]
@liranmauda you should not move this to draft, we currently still plan to use it. Also when postgresql is available with noobaa we then would like to provide our own postgresql instance and specify our credentials via a secret..

**Comment by nimrod-becker** [comment]
Closing this stale, PR.
As we discussed we also want to start moving away from MongoDB in upstream as well

**Pull request closed without merging**